segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentContainer.java

```diff
@@ -25,4 +24,13 @@
      * @return A Collection with the SegmentProperties for these segments.
      */
     Collection<SegmentProperties> getActiveSegments();
+
+    /**
+     * Gets a registered SegmentContainerPlugin of the given type.
+     *
+     * @param pluginClass Class of the plugin.
+     * @param <T>         Type of the plugin.
+     * @return A registered SegmentContainerPlugin of the requested type.
+     */
+    <T extends SegmentContainerPlugin> T getPlugin(Class<T> pluginClass);
```

Comment:
Isn't the idea the plugin is running internally to the container? If so vending it to the outside seems to break encapsulation. Why is this needed?

Reply:
FYI: at @fpj 's suggestion I renamed `SegmentContainerPlugin` to `SegmentContainerExtension` to make it clear this is not something pluggable via some configuration or dynamically at runtime.